src/lib.rs

```diff
@@ -237,6 +278,53 @@ impl<T: Columnation> Default for ColumnStack<T> {
     }
 }
 
+impl<T: Columnation, B: Borrow<T>> Extend<B> for ColumnStack<T> {
+    fn extend<I: IntoIterator<Item=B>>(&mut self, iter: I) {
+        for element in iter {
+            self.copy(element.borrow())
+        }
+    }
+}
+
+impl<A: Columnation, B: Borrow<A>> std::iter::FromIterator<B> for ColumnStack<A> {
+    fn from_iter<T: IntoIterator<Item=B>>(iter: T) -> Self {
+        let iter = iter.into_iter();
+        let mut c = ColumnStack::<A>::with_capacity(iter.size_hint().0);
+        c.extend(iter);
+        c
+    }
+}
```

Comment:
Nit, but elsewhere `T` is the `Columnation` type. Not always, now looking at it, but often.

Reply:
Fixed, and removed the `Borrow` requirement. It turns out it's difficult to use this way.